packages/tools-react-native/src/platform.ts

```diff
@@ -134,22 +169,58 @@ export function platformExtensions(platform: string): string[] {
   }
 }
 
+/**
+ * @returns the given string as a platform value or undefined if it is not a valid platform.
+ */
+export function parsePlatformValue(val: string): AllPlatforms | undefined {
+  return allValues.includes(val as AllPlatforms)
+    ? (val as AllPlatforms)
+    : undefined;
+}
+
 /**
  * Parse a string to ensure it maps to a valid react-native platform.
  *
  * @param val Input string
  * @returns React-native platform name. Throws `Error` on failure.
  */
 export function parsePlatform(val: string): AllPlatforms {
-  switch (val) {
-    case "android":
-    case "ios":
-    case "macos":
-    case "win32":
-    case "windows":
-      return val;
-
-    default:
-      throw new Error("Invalid platform '" + val + "'");
+  const platform = parsePlatformValue(val);
+  if (!platform) {
+    throw new Error(`Invalid platform '${val}'`);
```

Comment:
We should change this to say unknown instead of invalid:
```suggestion
    throw new Error(`Unknown platform: ${val}`);
```

Reply:
Done